web/packages/teleterm/src/ui/DocumentTerminal/Terminal/Terminal.tsx

```diff
@@ -83,9 +110,20 @@ export function Terminal(props: TerminalProps) {
       width="100%"
       style={{ overflow: 'hidden' }}
     >
+      {startPtyProcessAttempt.status === 'error' && (
```

Comment:
Why handle this in `Terminal` rather than in `DocumentTerminal` which already handles errors related to initializing the PTY process?

`Terminal` is the component responsible for calling `start` on the PTY process. I tried to have this logic in `DocumentTerminal` at first, but I ran into some weird edge cases when reattempting the call to initialize a PTY process.

If the logic is contained in Terminal, then we know that we're going to call `start` only once for each PTY and that when a new PTY is created, Terminal is going to have its state reset.